### PR TITLE
fix: search params handling nulls

### PR DESCRIPTION
**Pull request by aakropotkin**
Fixes parsing of `null` fields in `pkgdb search` params to align with spec.

### CODE REVIEW

File: tests/search-params.cc [hunk 1]
```diff
@@ -22,18 +22,14 @@ using namespace nlohmann::literals;
...
-            << "      stabilities: "
-            << nlohmann::json(
-                 params.stabilities.value_or( std::vector<std::string> {} )
-               ).dump() << std::endl;
+/**
+ * @brief Ensure defaults/fallbacks work correctly with
+ *        @a flox::search::SearchParams `from_json`.
+ */
+  bool
+test_SearchParams_defaults0() {
```
Comment (zmitchell) [inline tests/search-params.cc:30]: What is the point of this function if it always returns true? Also, is it ever used?

**Comment by zmitchell** [comment]
This seems to check whether several parameters are `null` when the spec doesn't say that they should be allowed to be `null` e.g. `allow`, `registry`, etc

File: tests/search-params.cc (hunk 1 above)
Comment (aakropotkin) [reply tests/search-params.cc:30]: Ah I forgot to remove it.

This was a test of the review system, and you passed 😬

**Comment by zmitchell** [comment]
Once all the lights are green I'll approve and merge

**Comment by aakropotkin** [comment]
> This seems to check whether several parameters are `null` when the spec doesn't say that they should be allowed to be `null` e.g. `allow`, `registry`, etc

For historical record:
We discussed this in more detail on a call. The summary is
> The parser is what is being tested here, the underlying query runner is responsible for throwing errors if a registry or other mandatory fields are not provided.
> The separation exists because internally some interfaces like `search` require a `registry` to be provided explicitly, while other like `resolver` ( specifically with a _manifest_ ) do not require these to be set at this layer.
> Various layers of base classes exist to adhere to DRY, and because of that the `search` params parser layer allows these fields to be omitted; but if you try to execute the query it'll blow up.
> From an API user's perspective with `pkgdb search` the field `registry` is required, but the parser relies on a base class to enforce this and generate error messages.

**Pull request merged**